AudioSinkExample/ViewController.swift

```diff
@@ -0,0 +1,654 @@
+            if (self.messageLabel.isHidden == false) {
+                UIView.animate(withDuration: 0.6, animations: {
+                    self.messageLabel.alpha = 0
+                }, completion: { (complete) in
+                    if (complete) {
+                        self.messageLabel.isHidden = true
+                    }
+                })
+            }
+        }
+
+        self.messageTimer = timer
+        RunLoop.main.add(timer, forMode: .commonModes)
+    }
+
+    // MARK: Speech Recognition
+    func stopRecognizingAudio() {
+        if let recognizer = self.speechRecognizer {
+            recognizer.stopRecognizing()
+            self.speechRecognizer = nil
+
+            if let view = self.speechRecognizerView {
+                hideSpeechRecognitionUI(view: view)
+            }
+        }
+    }
+
+    func recognizeRemoteAudio(gestureRecognizer: UIGestureRecognizer) {
+        guard let remoteView = gestureRecognizer.view else {
+            print("Couldn't find a view attached to the tap recognizer. \(gestureRecognizer)")
+            return;
+        }
+
+        // Find the Participant.
+        let hashedSid = remoteView.tag
+        for remoteParticipant in room!.remoteParticipants {
+            for videoTrackPublication in remoteParticipant.remoteVideoTracks {
+                if (videoTrackPublication.trackSid.hashValue == hashedSid) {
+                    if let audioTrack = remoteParticipant.remoteAudioTracks.first?.remoteTrack {
+                        recognizeRemoteParticipantAudio(audioTrack: audioTrack,
+                                                        sid: remoteParticipant.remoteAudioTracks.first!.trackSid,
+                                                        name: remoteParticipant.identity,
+                                                        view: remoteView)
+                    }
+                }
+            }
+        }
+    }
+
+    func recognizeRemoteParticipantAudio(audioTrack: TVIRemoteAudioTrack, sid: String, name: String, view: UIView) {
+        if (self.speechRecognizer != nil) {
+            stopRecognizingAudio()
+        } else {
+            showSpeechRecognitionUI(view: view, message: "Listening to \(name)...")
+
+            recognizeAudio(audioTrack: audioTrack, identifier: sid)
+        }
+    }
+
+    func recognizeLocalAudio() {
+        if (self.speechRecognizer != nil) {
+            stopRecognizingAudio()
+        } else if let audioTrack = self.localAudioTrack {
+            // TODO: CE - Only allow this operation when we are in a Room.
+
+            if let view = self.camera?.previewView {
+                showSpeechRecognitionUI(view: view,
+                                        message: "Listening to \(room?.localParticipant?.identity ?? "yourself")...")
+            }
+
+            recognizeAudio(audioTrack: audioTrack, identifier: audioTrack.trackId)
+        }
```

Comment:
Problem - local audio capture doesn't start until you join a Group Room or P2P Room with >= 1 Participants. One thing to do would be getting stats. If we have an entry for our LocalAudioTrack then allow the operation?

```
room?.getStats( (stats) {
  if (stats.count >= 1) {
     // Allow the operation.
  }
})
```